internal/cli/conn.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * MIT License
+ *
+ * Copyright (c)  2018 Kasun Vithanage
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cli
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/kasvith/kache/internal/protcl"
+)
+
+var c *cli
+
+type cli struct {
+	conn   net.Conn
+	reader *bufio.Reader
+}
+
+func (r *cli) parseResp() (string, error) {
+	buf, err := r.reader.ReadBytes('\n')
+	if err != nil {
+		if err == io.EOF {
+			return "", nil
+		}
+		return "", err
+	}
+
+	if err := protcl.EndWithCRLF(buf); err != nil {
+		return "", err
+	}
+
+	switch buf[0] {
+	case '$':
+		strs, err := protcl.ParseMultiBulkReply(r.reader, buf)
```

Comment:
hmm this will fail if server sent a reply with integer array...as its possible with RESP